samila/genimage.py

```diff
@@ -80,10 +80,16 @@ def generate(
         range1 = list(float_range(self.start, self.stop, self.step))
         range2 = list(float_range(self.start, self.stop, self.step))
         range_prod = list(itertools.product(range1, range2))
+        except_in_calc = False
         for item in range_prod:
             random.seed(self.seed)
-            self.data1.append(self.function1(item[0], item[1]).real)
-            self.data2.append(self.function2(item[0], item[1]).real)
+            try:
+                self.data1.append(self.function1(item[0], item[1]).real)
+                self.data2.append(self.function2(item[0], item[1]).real)
```

Comment:
Consider this scenario:

1. Function1 is defined at (x,y)
2. Function2 is undefined at (x,y)

Output:
`data1` and `data2` lengths are not equal

Expected Behavior:

Update `data1` and `data2` if and only if `function1` and `function2` both are defined at that point.


**I think we need to move this part to a separate function**

Reply:
Solved in [this](https://github.com/sepandhaghighi/samila/pull/125/commits/58a26dc5c5785a723c7bc6fc74f5af221f2dbc3b) commit.